### PR TITLE
Initial Docker container

### PR DESCRIPTION
**Pull request by niclashedam**
The Dockerfile will create a docker container that is equivalent to running the _cijoe_ command. I have created a docker repo on [DockerHub](https://hub.docker.com/repository/docker/cijoe/cijoe). We should setup travis to push to there when releases are created.

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -16,7 +16,7 @@ configure: clean
 
 .PHONY: kmdo
 kmdo:
-	kmdo src
+	kmdo src || exit 0
```
Comment (safl) [inline docs/Makefile]: Please split this into its own commit w.g. "docs:build: osx related work-around"

File: docs/src/docker.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  docker run -it \
+  -v $(pwd)/target_env.sh:/cijoe/target_env.sh \
+  -v $(pwd)/results:/results \
+  cijoe
+
+You can now run your testplan. See `QuickStart`_ for examples on how to use
+**cijoe**. Starting the docker container is equivalent to running the **cijoe**
+command.
+
+.. note:: If you run **cijoe** on a remote system, bear in mind that you may
```
Comment (safl) [inline docs/src/docker.rst]: Would be great with documentation on how to transfer keys and manage populate the ssh-agent inside the container.

File: docs/Makefile (hunk 1 above)
Comment (niclashedam) [reply docs/Makefile]: Moved into a related OSX commit on #7.

File: docs/src/docker.rst (hunk 2 above)
Comment (niclashedam) [reply docs/src/docker.rst]: Alright. I have added a small feature that eases the pain of adding the keys to the docker container. It is not so easy to do for others due to the strict permission requirements. I have made the feature such that you can mount your SSH folder to `/tmp/cijoe/.ssh`, and then cijoe will do the rest.

I have also added a warning to the docs saying that this is insecure with untrused code, as the software has full access to the private key.

File: bin/cijoe [hunk 3]
```diff
@@ -47,6 +47,14 @@ if [[ ! -f "$ENV_FPATH" ]]; then
 		exit
 	fi
 fi
+
+# check if we are in docker
+if [ -f /.dockerenv ]; then
```
Comment (safl) [inline bin/cijoe]: Please use the ``[[`` ``]`` bash-ism

Comment (safl) [reply bin/cijoe]: Nevermind, I'll pull it in and edit that.
